201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.json

```diff
@@ -0,0 +1,254 @@
+        "Include",
+        "Exclude"
+      ],
+      "defaultValue": "Include",
+      "metadata": {
+        "description": "If enabled the provisioned Jenkins instance will include a pipeline that builds and deploys a Docker image to an Azure Container Registry. The fields below are required only when Include is selected."
+      }
+    },
+    "servicePrincipalClientId": {
+      "type": "string",
+      "metadata": {
+        "description": "Required when including the 'Push to ACR Pipeline'. Service Principal Client ID (also called App ID) used by Jenkins to push the docker image to Azure Container Registry."
+      },
+      "defaultValue": ""
+    },
+    "servicePrincipalClientSecret": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Required when including the 'Push to ACR Pipeline'. Service Principal secret used by Jenkins to push the docker image to Azure Container Registry."
+      },
+      "defaultValue": ""
+    },
+    "gitRepository": {
+      "type": "string",
+      "metadata": {
+        "description": "Required when including the 'Push to ACR Pipeline'. URL to a public git repository that includes a Dockerfile."
+      },
+      "defaultValue": "https://github.com/lwander/spin-kub-demo.git"
+    }
+  },
+  "variables": {
+    "resourcePrefix": "jenkins",
+    "storageAccountName": "[concat(variables('resourcePrefix'), uniquestring(resourceGroup().id))]",
+    "acrStorageAccountName": "[concat('registry', uniqueString(resourceGroup().id))]",
+    "acrName": "[uniqueString(resourceGroup().id)]",
+    "OSDiskName": "[concat(variables('resourcePrefix'), 'OSDisk')]",
+    "nicName": "[concat(variables('resourcePrefix'), 'VMNic')]",
+    "subnetName": "[concat(variables('resourcePrefix'), 'Subnet')]",
+    "publicIPAddressName": "[concat(variables('resourcePrefix'), 'PublicIP')]",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "[concat(variables('resourcePrefix'), 'VM')]",
+    "virtualNetworkName": "[concat(variables('resourcePrefix'), 'VNET')]",
+    "vmExtensionName": "initializeJenkins",
+    "vmSize": "Standard_DS1_v2",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/master/jenkins/",
+    "_extensionScript": "install-jenkins-and-docker-build-pipeline.sh",
+    "_artifactsLocationSasToken": ""
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2016-01-01",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "Storage",
+      "properties": {}
+    },
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('acrStorageAccountName')]",
+      "apiVersion": "2016-01-01",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "Storage",
+      "properties": {}
```

Comment:
this is the VM's storage account. I just changed the ACR storage account definition to be encrypted.